support-frontend/test/selenium/util/IdapiBypassRecaptchaRequest.scala

```diff
@@ -0,0 +1,54 @@
+package selenium.util
+
+import okhttp3.{OkHttpClient, Request, RequestBody, MediaType, Response}
+import io.circe.generic.semiauto.deriveDecoder
+import io.circe.{Json, Decoder, DecodingFailure}
+import io.circe.parser.parse
+
+import org.slf4j.LoggerFactory
+
+object IdapiBypassRecaptchaRequest {
+
+  private val client = new OkHttpClient()
+
+  private def logger = LoggerFactory.getLogger(this.getClass)
+
+  private val request: Request = new Request.Builder()
+    .url(Config.idapiNewTestUserUrl)
+    .addHeader(Config.idapiClientAccessTokenName, Config.idapiClientAccessTokenSecret)
+    .post(RequestBody.create(MediaType.parse("application/json"), "{}"))
+    .build()
+
+  private def getResult(): Either[String, Json] = {
+    val response: Response = client.newCall(request).execute()
+    val body: String = response.body().string()
+    parse(body) match {
+      case Left(error) => Left(s"Could not parse $body as JSON from response with code ${response.code()}: $error")
+      case Right(json) => Right(json)
+    }
+  }
+
+  def getCookies(): Either[String, List[IdapiCookie]] = {
+    getResult() match {
+      case Left(error) => Left(error)
+      case Right(result) => {
+        result.as[IdapiSuccessfulResponse] match {
+          case Left(error) => Left(s"Could not deserialise IDAPI response: $error")
+          case Right(r) => Right(r.values)
+        }
+      }
+    }
```

Comment:
when using a Either or other specific error type, you can use a for comprehension get rid of most of the boilerplate but maintain the error in the background
```suggestion
    for {
      result <- getResult()
      deserialised <- result.as[IdapiSuccessfulResponse].leftMap(error => s"Could not deserialise IDAPI response: $error")
    } yield deserialised.values
```

Reply:
Will read up on for comprehension now

Reply:
I had a look at this one for a bit and I'm still not sure how this particular for comprehension gets resolved. Leaving it out for a future refactoring PR.